taxi_demo/etl/etl.ipynb

```diff
@@ -0,0 +1,938 @@
+    "            'dropoff_datetime',\n",
+    "            'passenger_count',\n",
+    "            'trip_distance',\n",
+    "            'rate_code_id',\n",
+    "            'store_and_fwd_flag',\n",
+    "            'pickup_taxizone_id',\n",
+    "            'dropoff_taxizone_id',\n",
+    "            'payment_type',\n",
+    "            'fare_amount',\n",
+    "            'extra',\n",
+    "            'mta_tax',\n",
+    "            'tip_amount',\n",
+    "            'tolls_amount',\n",
+    "            'improvement_surcharge',\n",
+    "            'total_amount',\n",
+    "            'junk1',  # extra trailing commas\n",
+    "            'junk2',\n",
+    "        ]\n",
+    "    },\n",
+    "    '2017_2018': {\n",
+    "        'date_range': ('2017-01', '2019-01'),\n",
+    "        'cols': [\n",
+    "            'vendor_id',\n",
+    "            'pickup_datetime',\n",
+    "            'dropoff_datetime',\n",
+    "            'passenger_count',\n",
+    "            'trip_distance',\n",
+    "            'rate_code_id',\n",
+    "            'store_and_fwd_flag',\n",
+    "            'pickup_taxizone_id',\n",
+    "            'dropoff_taxizone_id',\n",
+    "            'payment_type',\n",
+    "            'fare_amount',\n",
+    "            'extra',\n",
+    "            'mta_tax',\n",
+    "            'tip_amount',\n",
+    "            'tolls_amount',\n",
+    "            'improvement_surcharge',\n",
+    "            'total_amount',\n",
+    "        ]\n",
+    "    },\n",
+    "    '2019': {\n",
+    "        'date_range': ('2019-01', '2020-01'),\n",
+    "        'cols': [\n",
+    "            'vendor_id',\n",
+    "            'pickup_datetime',\n",
+    "            'dropoff_datetime',\n",
+    "            'passenger_count',\n",
+    "            'trip_distance',\n",
+    "            'rate_code_id',\n",
+    "            'store_and_fwd_flag',\n",
+    "            'pickup_taxizone_id',\n",
+    "            'dropoff_taxizone_id',\n",
+    "            'payment_type',\n",
+    "            'fare_amount',\n",
+    "            'extra',\n",
+    "            'mta_tax',\n",
+    "            'tip_amount',\n",
+    "            'tolls_amount',\n",
+    "            'improvement_surcharge',\n",
+    "            'total_amount',\n",
+    "            'congestion_surcharge',\n",
+    "        ]\n",
+    "    }\n",
+    "}\n",
+    "\n",
+    "def get_files(date_range):\n",
+    "    return [f'{base_path}_{x}.csv' \n",
+    "            for x in pd.date_range(date_range[0], date_range[1], freq='M').strftime(\"%Y-%m\")]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def load_csv(date_range, cols):\n",
+    "    df = dd.read_csv(\n",
+    "        get_files(date_range),\n",
+    "        header=0,\n",
+    "        na_values=[\"NA\"],\n",
+    "        parse_dates=[1, 2],  # dates are in same position for each file\n",
```

Comment:
Positions are used because we deliberately ignore the column names, as the column names changed a few times over the years (thanks NYC). I could include into the json spec what the column names are for each file but I don't think that's worth the effort at this point